src/client/method-is-enabled.c

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright Contributors to the Eclipse BlueChi project
+ *
+ * SPDX-License-Identifier: LGPL-2.1-or-later
+ */
+#include <stdlib.h>
+#include <string.h>
+
+#include "client.h"
+#include "method-is-enabled.h"
+#include "usage.h"
+
+#include "libbluechi/common/opt.h"
+
+static int method_is_enabled_on(Client *client, char *node_name, char *unit_file) {
+        int r = 0;
+        _cleanup_sd_bus_error_ sd_bus_error error = SD_BUS_ERROR_NULL;
+        _cleanup_sd_bus_message_ sd_bus_message *result = NULL;
+        const char *state = NULL;
+
+        r = assemble_object_path_string(NODE_OBJECT_PATH_PREFIX, node_name, &client->object_path);
+        if (r < 0) {
+                return r;
+        }
+
+        r = sd_bus_call_method(
+                        client->api_bus,
+                        BC_INTERFACE_BASE_NAME,
+                        client->object_path,
+                        NODE_INTERFACE,
+                        "GetUnitFileState",
+                        &error,
+                        &result,
+                        "s",
+                        unit_file);
+        if (r < 0) {
+                fprintf(stderr, "Failed to issue method call: %s\n", error.message);
+                return r;
+        }
+
+        r = sd_bus_message_read(result, "s", &state);
+        if (r < 0) {
+                fprintf(stderr, "Failed to read result of method call: %s\n", error.message);
+                return r;
+        }
+
+        printf("%s\n", state);
+
+        if (strcmp(state, "enabled") == 0 || strcmp(state, "enabled-runtime") == 0 ||
```

Comment:
All we are doing here is sending a request to the systemd.
So if systemd adds a new state we will need to rewrite this if.
So maybe it's better to check if it's not NULL and if it's not return 0?

Reply:
No, we want to provide the same functionality of `systemctl is-enabled`, which means we should print the current enablement status of a unit file and return `0` if unit file is enabled (that's the condition which is aligned with systemctl) or non-zero otherwise.